### ISSUE TITLE
Generated equals doesn't repect nested classes

### ISSUE THREAD

**Issue by snclever** [REggar/auto-value-ignore-hash-equals#8]
When you have a nested class annotated with `@AutoValue` and `@IgnoreHashEquals` or `@IncludeHashEquals` the generated code doesn't compile.

Example:

```
public class Parent {
    @AutoValue
    static abstract class Test {
        abstract String a();
        @IgnoreHashEquals
        abstract String b();
    }
}
```

generates

```
final class AutoValue_Parent_Test extends $AutoValue_Parent_Test {
  AutoValue_Parent_Test(String a, String b) {
    super(a, b);
  }

  @Override
  public final boolean equals(Object o) {
    if (o == this) {
      return true;
    }
    if (o instanceof Test) {
      Test that = (Test) o;
      return (this.a().equals(that.a()));
    }
    return false;
  }

  @Override
  public final int hashCode() {
    int h = 1;
    h *= 1000003;
    h ^= this.a().hashCode();
    return h;
  }
}
```

The problematic part is `... if (o instanceof Test) { Test that = (Test) o; ...` which has to be  `... if (o instanceof Parent.Test) { Parent.Test that = (Parent.Test) o; ...`.

@REggar I will make a pull request fixing this issue if you don't object.

**Comment by REggar** [comment]
Thanks for raising the issue, if you could please go ahead with the pull request that would be great. Let me know if you get stuck and I'll have a look.

**Comment by REggar** [comment]
Merged in release 1.1.3, which has been published and (should) be available within the next couple of hours.

Thanks again for your help.

EDIT: It's now available.
